### PR TITLE
Fix .webp SrcSet Creation Filter

### LINKED ISSUE #2777: Responsive Image SrcSet Creation Not Functioning Properly
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

In PR #2698, a filter was added for creating the `.webp` srcset for responsive images. However, the filter currently searches for ".jpg" and ".png" instead of "jpg" and "png", leading to no matches. This issue arises because the split operation removes the period from the extension, preventing the srcset from being generated for any figures. The filter needs to be updated to ensure that responsive image srcsets are created correctly for the appropriate file extensions.

### How to reproduce the bug

Include a Picture (File Ending ".jpg"), enable imagemagick and check the produced generated source code (HTML) how the images are included

### Error messages and logs

_No response_

### What operating system are you using?

Mac

### Where are you seeing the problem on?

Deployed site

### More info

_No response_

### PR DESCRIPTION
**Pull request by dfuchss**
#### Description

This PR addresses an issue with the filter for creating `.webp` srcsets introduced in PR #2698. The original filter incorrectly searched for extensions with a leading period (e.g., ".jpg" and ".png"). As a result, no matches were found, preventing the srcset from being added to any figures. This occurred because the split operation removes the period from the file extensions.

#### Changes Made

- Updated the filter to search for file extensions without the leading period (e.g., "jpg" and "png").
- Ensured that the resource sets are now correctly created for the respective file extensions, allowing the `.webp` srcset to function as intended.

#### Impact

With this change, responsive images will now correctly generate their srcsets for the appropriate file types, improving image loading performance and supporting better responsiveness.

#### Fixes

- Fixes #2777

Please review the changes and let me know if there are any further adjustments needed. Thank you!

### CODE REVIEW

**Comment by nWestie** [comment]
^^ This - I came to fix this same issue, and found someone already got to it! simple fix.

File: _includes/figure.liquid [hunk 1]
```diff
@@ -16,7 +16,7 @@
     {% if site.imagemagick.enabled %}
       <source
         class="responsive-img-srcset"
-        {% if ext == '.jpg' or ext == '.jpeg' or ext == '.png' or ext == '.tiff' or ext == '.gif' %}
+        {% if ext == 'jpg' or ext == 'jpeg' or ext == 'png' or ext == 'tiff' or ext == 'gif' %}
```
Comment (george-gca) [inline _includes/figure.liquid:19]: Thanks for the fix. Can you also move the extensions to be in alphabetical order?

**Review by george-gca — CHANGES_REQUESTED** [review]

File: _includes/figure.liquid (hunk 1 above)
Comment (dfuchss) [reply _includes/figure.liquid:19]:
```suggestion
        {% if ext == 'gif' or ext == 'jpeg' or ext == 'jpg' or ext == 'png' or ext == 'tiff' %}​⬤
```

File: _includes/figure.liquid [hunk 2]
```diff
@@ -16,7 +16,7 @@
     {% if site.imagemagick.enabled %}
       <source
         class="responsive-img-srcset"
-        {% if ext == '.jpg' or ext == '.jpeg' or ext == '.png' or ext == '.tiff' or ext == '.gif' %}
+        {% if ext == 'gif' or ext == 'jpeg' or ext == 'jpg' or ext == 'png' or ext == 'tiff' %}​⬤
```
Comment (dfuchss) [inline _includes/figure.liquid:19]:
```suggestion
        {% if ext == 'gif' or ext == 'jpeg' or ext == 'jpg' or ext == 'png' or ext == 'tiff' %}​
```

**Comment by george-gca** [comment]
Just missing the prettier fix.

**Comment by dfuchss** [comment]
> Just missing the prettier fix.

I don't know whether prettier is requesting this because I only changed one line. But nevertheless, it's applied now.

**Pull request merged**